rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -212,4 +213,120 @@ Node::create_service(
   return serv;
 }
 
+const std::vector<rcl_interfaces::SetParametersResult>
+Node::set_parameters(
+  const std::vector<rcl_interfaces::Parameter> & parameters)
+{
+  std::vector<rcl_interfaces::SetParametersResult> results;
+  for (auto p : parameters) {
+    parameters_[p.name] = rclcpp::parameter::ParameterVariant::from_parameter(p);
+    rcl_interfaces::SetParametersResult result;
+    result.successful = true;
+    // TODO: handle parameter constraints
+    results.push_back(result);
+  }
+  return results;
+}
+
+const rcl_interfaces::SetParametersResult
+Node::set_parameters_atomically(
+  const std::vector<rcl_interfaces::Parameter> & parameters)
+{
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
@esteve can you explain a little bit why we need a mutex here but not in the other functions?